### PR TITLE
Item viewholders

### PR DESCRIPTION
**Pull request by Zsteven44**

### CODE REVIEW

File: app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java [hunk 1]
```diff
@@ -9,11 +9,29 @@
...
 import tech.zafrani.pubgapp.models.Item;
 
 
 public class ItemListAdapter extends RecyclerView.Adapter<ItemViewHolder> {
+    private static final String ARG_WEAPONS = "Weapons";
+    private static final String ARG_AMMUNITION = "Ammunition";
+    private static final String ARG_ATTACHMENTS = "Attachments";
+    private static final String ARG_EQUIPMENT = "Equipment";
+    private static final String ARG_CONSUMABLES = "Consumables";
+    private static final String ARG_VEHICLES = "Vehicles";
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java]: These should not be args

File: app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java [hunk 2]
```diff
@@ -9,11 +9,29 @@
...
+    private static final int VAL_WEAPONS = 0;
+    private static final int VAL_AMMUNITION = 1;
+    private static final int VAL_ATTACHMENTS = 2;
+    private static final int VAL_EQUIPMENT = 3;
+    private static final int VAL_CONSUMABLES = 4;
+    private static final int VAL_VEHICLES = 5;
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java]: These shouldn't be vals, they should be ID's

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/AmmunitionViewHolder.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    private final TextView nameTextView;
+    private final TextView typeTextView;
+    private final TextView categoryTextView;
+    private final TextView capacityTextView;
+    private final ImageView imageImageView;
+    private final Context context;
+
+    public AmmunitionViewHolder(@NonNull final View itemView, @NonNull final Context context) {
+        super(itemView);
+        this.context = context;
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/AmmunitionViewHolder.java]: Don't need context. Can use itemView.getContext().

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/AttachmentsViewHolder.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        }
+    }
+    private void setItemCapacity(@Nullable final Float itemCapacity) {
+        if (itemCapacity != null) {
+            this.capacityTextView.setText(context.getString(R.string.row_item_capacity, itemCapacity));
+        }else {
+            this.capacityTextView.setText(EMPTY_FIELD);
+        }
+    }
+    private void setItemDamage(@Nullable final HashMap<String,String> itemDamage) {
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/AttachmentsViewHolder.java]: This hashmap should be a class that extends HashMap<String,String> and provides a function `getChest0()`

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ConsumablesViewHolder.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    private final TextView maxHealTextView;
+    private final TextView boostTextView;
+    private final TextView activateTimeTextView;
+    private final TextView maxStackTextView;
+    private final ImageView imageImageView;
+    private final Context context;
+
+    public ConsumablesViewHolder(@NonNull final View itemView, @NonNull final Context context) {
+        super(itemView);
+        this.context = context;
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ConsumablesViewHolder.java]: This context isn't needed

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/EquipmentViewHolder.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    private final TextView categoryTextView;
+    private final TextView capacityTextView;
+    private final TextView protectionTextView;
+    private final ImageView imageImageView;
+    private final Context context;
+
+    public EquipmentViewHolder(@NonNull final View itemView,
+                               @NonNull final Context context) {
+        super(itemView);
+        this.context = context;
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/EquipmentViewHolder.java]: this context isn't needed

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/VehicleViewHolder.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    private final TextView accelerationTextView;
+    private final TextView healthTextView;
+    private final TextView seatsTextView;
+    private final ImageView imageImageView;
+    private final Context context;
+
+    public VehicleViewHolder(@NonNull final View itemView,
+                             @NonNull final Context context) {
+        super(itemView);
+        this.context = context;
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/VehicleViewHolder.java]: this context isn't needed

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/WeaponsViewHolder.java [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+        }
+    }
+    private void setItemCapacity(@Nullable final Float itemCapacity) {
+        if (itemCapacity != null) {
+            this.capacityTextView.setText(context.getString(R.string.row_item_capacity, itemCapacity));
+        }else {
+            this.capacityTextView.setText(EMPTY_FIELD);
+        }
+    }
+    private void setItemDamage(@Nullable final HashMap<String,String> itemDamage) {
```
Comment (zafrani) [inline app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/WeaponsViewHolder.java]: Should use the HashMap object I mentioned above.

File: app/src/main/res/layout/row_item_ammunition.xml [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    >
+    <LinearLayout
+        android:orientation="horizontal"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginBottom="8dp"
```
Comment (zafrani) [inline app/src/main/res/layout/row_item_ammunition.xml]: should use small_margin

File: app/src/main/res/layout/row_item_ammunition.xml [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+                    android:layout_weight="0.20">
+
+                </LinearLayout>
+            </LinearLayout>
+        </LinearLayout>
+    </LinearLayout>
+    <View
+    android:layout_height="1dp"
+    android:layout_width="match_parent"
+    android:background="@color/colorDivider">
```
Comment (zafrani) [inline app/src/main/res/layout/row_item_ammunition.xml]: should just be divider

File: app/src/main/res/layout/row_item_attachment.xml [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    >
+    <LinearLayout
+        android:orientation="horizontal"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginBottom="8dp"
```
Comment (zafrani) [inline app/src/main/res/layout/row_item_attachment.xml]: should use small_margin

File: app/src/main/res/layout/row_item_equipment.xml [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    >
+    <LinearLayout
+        android:orientation="horizontal"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginBottom="8dp"
```
Comment (zafrani) [inline app/src/main/res/layout/row_item_equipment.xml]: should use small_margin

File: app/src/main/res/layout/row_item_weapon.xml [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+    <LinearLayout
+        android:orientation="horizontal"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginBottom="8dp"
```
Comment (zafrani) [inline app/src/main/res/layout/row_item_weapon.xml]: should use small_margin

**Review by zafrani — CHANGES_REQUESTED** [review]

File: app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java (hunk 1 above)
Comment (Zsteven44) [reply app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java]: done

File: app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java (hunk 2 above)
Comment (Zsteven44) [reply app/src/main/java/tech/zafrani/pubgapp/adapters/ItemListAdapter.java]: done

File: app/src/main/res/layout/row_item_ammunition.xml (hunk 9 above)
Comment (Zsteven44) [reply app/src/main/res/layout/row_item_ammunition.xml]: done

File: app/src/main/res/layout/row_item_ammunition.xml (hunk 10 above)
Comment (Zsteven44) [reply app/src/main/res/layout/row_item_ammunition.xml]: done

File: app/src/main/res/layout/row_item_attachment.xml (hunk 11 above)
Comment (Zsteven44) [reply app/src/main/res/layout/row_item_attachment.xml]: done

File: app/src/main/res/layout/row_item_equipment.xml (hunk 12 above)
Comment (Zsteven44) [reply app/src/main/res/layout/row_item_equipment.xml]: done

File: app/src/main/res/layout/row_item_weapon.xml (hunk 13 above)
Comment (Zsteven44) [reply app/src/main/res/layout/row_item_weapon.xml]: done

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/VehicleViewHolder.java (hunk 7 above)
Comment (Zsteven44) [reply app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/VehicleViewHolder.java]: done

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/EquipmentViewHolder.java (hunk 6 above)
Comment (Zsteven44) [reply app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/EquipmentViewHolder.java]: done

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ConsumablesViewHolder.java (hunk 5 above)
Comment (Zsteven44) [reply app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/ConsumablesViewHolder.java]: done

File: app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/AmmunitionViewHolder.java (hunk 3 above)
Comment (Zsteven44) [reply app/src/main/java/tech/zafrani/pubgapp/adapters/viewholders/AmmunitionViewHolder.java]: done

**Review by zafrani — APPROVED** [review]

**Pull request merged**